tm2py/components/network/transit/transit_assign.py

```diff
@@ -2231,13 +2231,41 @@ def _journey_levels(self) -> EmmeTransitJourneyLevelSpec:
 
             elif self.name == "WLK_TRN_WLK":
                 new_journey_levels = copy.deepcopy(journey_levels)
-                transition_rules = copy.deepcopy(journey_levels[0]["transition_rules"])
+
+                for i in range(0, len(new_journey_levels)):
+                    jls = new_journey_levels[i]
+                    jls["transition_rules"].extend(
+                        [
+                            {"mode": "e", "next_journey_level": i+1},
+                            {"mode": "w", "next_journey_level": i+1},
+                            {
+                                "mode": "a",
+                                "next_journey_level": i+1,
+                            },
+                        ]
+                    )
+                # level 0: only allow walk access and walk auxilary
+                # must use the trasit modes to get onto the next level, 
+                transition_rules_walk = copy.deepcopy(journey_levels[0]["transition_rules"])
+                transition_rules_walk.extend(
+                    [
+                        {
+                            "mode": "e",
+                            "next_journey_level": 0,
+                        },
+                        {
+                            "mode": "w",
+                            "next_journey_level": 0,
+                        },
+                        {"mode": "a", "next_journey_level": 0},
+                    ]
+                )
                 new_journey_levels.insert(
```

Comment:
At this point we have updated the `journey_level` list to include a new `journey_level` 0 and allowed the ability to transition out of this state to any of the other states. Now we want to attribute the the level 0 state. The key attribute is `destinations_reachable` set to `False`. This disallows walk-all-the-way, as you have to transition out of this state to reach your destination. To transition out of the state, you need to board a transit vehicle. The `transition_rules`, which are defined above, define what happens when a traveler goes from their existing mode to `w`, `a`, or `e` (walk auxiliary, walk access, or walk egress), which is they remain in state 0. Again, to get you of state 0, you must board a transit vehicle and enter one of the other states. 

fyi @lmz, @FlaviaTsang